helm/flink-kubernetes-operator/templates/rbac.yaml

```diff
@@ -21,23 +21,14 @@ RBAC rules used to create the operator (cluster)role based on the scope
 */}}
 {{- define "flink-operator.rbacRules" }}
 rules:
-  - apiGroups:
-      - flink-operator
-    resources:
-      - "*"
-    verbs:
-      - "*"
   - apiGroups:
       - ""
     resources:
       - pods
       - services
-      - endpoints
-      - persistentvolumeclaims
       - events
       - configmaps
       - secrets
-      - nodes
```

Comment:
I remember we needed it either when playing with Ingress or the `kubernetes.rest-service.exposed.type` fail to recall. Worth double-checking both before removing.

Reply:
I will double check that config and get back.

Reply:
@morhidi You are right, `kubernetes.rest-service.exposed.type: NodePort` requires `list` permission on `nodes` at the cluster scope.

```
2022-07-01 13:30:20,742 i.j.o.p.e.ReconciliationDispatcher [ERROR][flink-27975/basic-example] Error during event processing ExecutionScope{ resource id: ResourceID{name='basic-example', namespace='flink-27975'}, version: 16843355} failed.
org.apache.flink.kubernetes.operator.exception.ReconciliationException: org.apache.flink.kubernetes.shaded.io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://10.96.0.1/api/v1/nodes. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. nodes is forbidden: User "system:serviceaccount:flink-27975:flink-operator" cannot list resource "nodes" in API group "" at the cluster scope.
```

This brings another point that `kubernetes.rest-service.exposed.type: NodePort` will throw error for namespace scoped operator as it requires `list` permission at the cluster scope for `nodes`.

I will update the PR with `nodes` rbac and some comments. Thanks for your pointers.

@gyfora FYI